src/lex/cpp.rs

```diff
@@ -1647,7 +1647,9 @@ int main(){}
 
     #[test]
     fn space_separated_function_macro() {
-        let src = "#define f(a) <a>\nf     (a)";
-        assert_same(src, "<a>");
+        assert_same_exact("#define f(a) <a>\nf     (a)", "\n<a>");
+        assert_same_exact("#define f(a) <a>\nf(a)", "\n<a>");
+        assert_same_exact("#define f <a>\nf", "\n<a>");
+        assert_same_exact("#define f <a>\nf   ", "\n<a>   ");
```

Comment:
This doesn't test the right thing.
```suggestion
        assert_same_exact("#define f(a) <a>\nf", "\nf");
        assert_same_exact("#define f(a) <a>\nf   ", "\nf   ");
```

Reply:
Maybe add one more with something other than a parenthesis or eof after. E.g.
```
#define f(a) <a>
f   ;
```